### DISCUSSION TITLE
Multiple App-Instances with 1 SubProcess or separated SubProcesses?

### DISCUSSION THREAD

**Discussion by lkienberger** [cefsharp/CefSharp#4924] [General]
Dear community!
I have an WPF app, where user can start multiple instances and i already read that Cef.Initialize() may only be called once - if I call it 2 times the second app crashes. I already tried to set different cache-patches on both instances, but still the same problem. Is there any solution to use CefSharp-Subprocess for more than 1 instance of my app? E.g. a global SubProcess? I already tried every suggestion i found at google but had no solution until now.

**Comment by amaitland** [comment]
> I already tried to set different cache-patches on both instances, but still the same problem.

If you need two different processes then separate cache paths is the only option. Double check your settings.

The other option is to take the same approach `Chrome` does and keep a single process with multiple windows.

**Reply by e-m-hh-com** [reply]
Why is this now the only option? in some old versions of CefSharp this was possible without setting a cache path. Or can you suggest a better approach to use CefSharp with the minimum cache size, that it should be cleared after closing the app? Somehow when my app closes, CefSharp.BrowserSubprocess.exe is not yet being killed therefore cache cannot be deleted in AppDomain.CurrentDomain.ProcessExit. Right now I check on app startup if there is old cache files and delete them asynchronous, but that is not optimal.

**Reply by amaitland** [reply]
> Why is this now the only option? in some old versions of CefSharp this was possible without setting a cache path.

`Upstream` changes. You can read about them at https://github.com/chromiumembedded/cef/issues/3609

> Right now I check on app startup if there is old cache files and delete them asynchronous, but that is not optimal.

There is a known `upstream` bug. See https://github.com/chromiumembedded/cef/issues/3726
